app/controllers/concerns/step_code_params_concern.rb

```diff
@@ -0,0 +1,43 @@
+module StepCodeParamsConcern
+  extend ActiveSupport::Concern
+
+  private
+
+  def step_code_params
```

Comment:
it could be possible we maintain separate part 3 and part 9 params. 
There are nearly a number of params shared though....